rust/lancedb/resources/remote-catalog-open-api.yaml

```diff
@@ -0,0 +1,284 @@
+          headers:
+            Location:
+              schema: {type: string}
+
+    delete:
+      tags: [ Table ]
+      summary: Delete all tables
+      parameters:
+        - $ref: '#/components/parameters/dbName'
+      responses:
+        204:
+          description: All tables deleted
+
+
+  /v1/databases/{dbName}/tables/{tableName}:
+    get:
+      tags: [Table]
+      summary: Get table metadata
+      parameters:
+        - $ref: '#/components/parameters/dbName'
+        - $ref: '#/components/parameters/tableName'
+      responses:
+        200:
+          description: Table metadata
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TableMetadata'
+
+    put:
+      tags: [Table]
+      summary: Update table data
+      parameters:
+        - $ref: '#/components/parameters/dbName'
+        - $ref: '#/components/parameters/tableName'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/TableUpdate'
+      responses:
+        200:
+          description: Update successful
+
+    delete:
+      tags: [ Table ]
+      summary: Delete a specific table
+      parameters:
+        - $ref: '#/components/parameters/dbName'
+        - $ref: '#/components/parameters/tableName'
+      responses:
+        204:
+          description: Table deleted
+
+  /v1/databases/{dbName}/tables/{tableName}/name:
+    put:
+      tags: [ Table ]
+      summary: Rename table
+      parameters:
+        - $ref: '#/components/parameters/dbName'
+        - $ref: '#/components/parameters/tableName'
+      requestBody:
+        content:
+          application/json:
+            schema:
+              type: object
+              required: [ newName ]
+              properties:
+                newName: { type: string }
+      responses:
+        204:
+          description: Table renamed
+
+
+components:
+  parameters:
+    dbName:
+      name: dbName
+      in: path
+      required: true
+      schema: {type: string}
+
+    tableName:
+      name: tableName
+      in: path
+      required: true
+      schema: {type: string}
+
+    startAfter:
+      name: startAfter
+      in: query
+      schema: {type: string}
+
+    limit:
+      name: limit
+      in: query
+      schema: {type: integer}
+
+  schemas:
+    DatabaseList:
+      type: object
+      properties:
+        databases:
+          type: array
+          items: {type: string}
+        nextToken: {type: string}
+
+    CreateDatabaseRequest:
+      type: object
+      required: [name, mode]
+      properties:
+        name: {type: string}
+        mode:
+          type: string
+          enum: [Create, ExistOk, Overwrite]
+        options:
+          type: object
+          additionalProperties: { type: string }
+
+    DatabaseMetadata:
+      type: object
+      properties:
+        name: {type: string}
+        tableCount: {type: integer}
```

Comment:
why returning table count here? This is not commonly seen elsewhere, is there a specific use case?